LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,256 @@
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithProcessBasedServerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithManagedContainerResponse>> CreateBuildWithManagedContainer(CreateBuildWithManagedContainerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithManagedContainerAsync(request);
+        }
+
+        public async Task<PlayFabResult<CreateBuildWithCustomContainerResponse>> CreateBuildWithCustomContainer(CreateBuildWithCustomContainerRequest request)
+        {
+            Console.WriteLine($"Starting deployment {request.BuildName} for titleId, regions  {string.Join(", ", request.RegionConfigurations.Select(x => x.Region))}");
+
+            return await PlayFabMultiplayerAPI.CreateBuildWithCustomContainerAsync(request);
+        }
+
+        public List<PlayFab.MultiplayerModels.Port> PortMapping()
+        {
+            List<PlayFab.MultiplayerModels.Port> ports = null;
+
+            foreach (var portList in settings.PortMappingsList)
+            {
+                ports = portList?.Select(x => new PlayFab.MultiplayerModels.Port()
+                {
+                    Name = x.GamePort.Name,
+                    Num = settings.RunContainer ? x.GamePort.Number : 0,
+                    Protocol = (ProtocolType)Enum.Parse(typeof(ProtocolType), x.GamePort.Protocol)
+                }).ToList();
+            }
+
+            return ports;
+        }
+
+        public string GetAssetFileNameFromPath(string filePath)
+        {
+            return System.IO.Path.GetFileName(filePath);
+        }
+
+        public async Task<PlayFabResult<GetAssetDownloadUrlResponse>> FileExistsInBlob(string filename)
+        {
+            GetAssetDownloadUrlRequest downloadRequest = new() { FileName = filename };
+
+            return await PlayFabMultiplayerAPI.GetAssetDownloadUrlAsync(downloadRequest);
+        }
+
+        public async Task CheckAssetFiles(string filename)
+        {
+           
+            var filevalidator = FileExistsInBlob(filename);
+
+            if (filevalidator.Result.Result == null)
+            {
+                GetAssetUploadUrlRequest request1 = new() { FileName = filename };
+
+                //TODO: log progress of asset upload
+                var uriResult = await PlayFabMultiplayerAPI.GetAssetUploadUrlAsync(request1);
+
+                if (uriResult.Error != null)
+                {
+                    Console.WriteLine(uriResult.Error.ErrorMessage);
+                }
+                var uri = new System.Uri(uriResult.Result.AssetUploadUrl);
+
+                var blockBlob = new CloudBlockBlob(uri);
```

Comment:
We also need to check the case when we put wrong URI so get error from blockBlob.UploadFromFileAsync(filename);
This case, we don't want to proceed because asset would not be uploaded to the blob storage so build cannot be created successfully.

Reply:
I added a try catch here, thanks!